provider/image.go

```diff
@@ -741,9 +744,45 @@ func processLogLine(msg string) (string, error) {
 	return info, nil
 }
 
-func configureDockerClient(configs map[string]string) (*client.Client, error) {
+// When `verify` is set, this function will check that the connection to the docker daemon works.
+// If it doesn't and no host was configured, it will try to connect to the user's docker daemon.
+func configureDockerClient(configs map[string]string, verify bool) (*client.Client, error) {
```

Comment:
We only call `configureDockerClient(config, true)` in production. We never call with `verify=false` except in testing. Can we simply remove the `verify` parameter?

This will ensure that we test the code path we care about *and* will reduce complexity.

Reply:
I introduced `verify=false` _for_ testing. In unit tests, we cannot verify connections because they don't actually exist.

An alternative would be to extract relevant parts into interface and mock it for tests. Probably cleaner but pretty involved.

Reply:
If thats the case, I would love a comment saying:
> `verify` is a testing affordance, and will always be `true` in production.

It seems like we can move the tests from `configureDockerClient` to `configureDockerClientInner` will little work and remove the extra parameter.

I leave it up to you whether to add the comment or change the test target.